src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -826,15 +847,29 @@ private InternalAggregations reduceAggregations(List<PercolateShardResponse> sha
             return null;
         }
 
+        InternalAggregations aggregations;
         if (shardResults.size() == 1) {
-            return shardResults.get(0).aggregations();
-        }
-
-        List<InternalAggregations> aggregationsList = new ArrayList<>(shardResults.size());
-        for (PercolateShardResponse shardResult : shardResults) {
-            aggregationsList.add(shardResult.aggregations());
+            aggregations = shardResults.get(0).aggregations();
+        } else {
+            List<InternalAggregations> aggregationsList = new ArrayList<>(shardResults.size());
+            for (PercolateShardResponse shardResult : shardResults) {
+                aggregationsList.add(shardResult.aggregations());
+            }
+            aggregations = InternalAggregations.reduce(aggregationsList, new ReduceContext(null, bigArrays, scriptService));
+        }
+        if (aggregations != null) {
+            List<SiblingReducer> reducers = shardResults.get(0).reducers();
+            if (reducers != null) {
+                List<InternalAggregation> newAggs = new ArrayList<>(Lists.transform(aggregations.asList(), Reducer.FUNCTION));
```

Comment:
Agreed, although ideally I would like to get rid of it since all we are doing is using it to cast a list of one type to a list of another type. I just haven't worked out how I can get rid of it yet. Any thoughts on that are welcome :)